improve multi-core performance of qupsample_bilinear2d

mingfeima opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74999
* #68328
* #72575
* #71903
* #69667
* __->__ #69601
* #69600
* #69599
* #69598
* #69517

Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

mingfeima committed:
improve multi-core performance of qupsample_bilinear2d

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"




[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

Xia-Weiwen commented:
Hi @jerryzh168 We have a few int8-related PRs targeting 1.11. And this is one of them. Could you prioritize to review these PRs? Thanks.

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima commented:
TODO: optimize contiguous memory format as well, some model might run UpSample prior to Conv2d.

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima commented:
Please follow the notes from #69600

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

dzdang commented:
@mingfeima  Can you please also rebase this stack after fixing the lint issue I mentioned above? Thanks

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima committed:
Update on "improve multi-core performance of qupsample_bilinear2d"


Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

[ghstack-poisoned]

mingfeima commented on aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:
```diff
@@ -2055,88 +2055,90 @@ void qupsample_bilinear2d_nhwc_kernel(
+      int64_t b{0}, h2{0}, w2{0};
+      data_index_init(begin, b, nbatch, h2, output_height, w2, output_width);
+
+      for (const auto i : c10::irange(begin, end)) {
```
suppress unused variable warning.

mingfeima commented on aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:
```diff
@@ -2055,88 +2055,90 @@ void qupsample_bilinear2d_nhwc_kernel(
+      data_index_init(begin, b, nbatch, h2, output_height, w2, output_width);
+
+      for (const auto i : c10::irange(begin, end)) {
+        (void)i; //Suppress unused variable warning
```
Suppress unused variable warning

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).
